src/main/java/backend/control/RepoOpControl.java

```diff
@@ -10,23 +12,43 @@
 import java.util.HashMap;
 import java.util.List;
 import java.util.Map;
+import java.util.Optional;
 import java.util.concurrent.*;
 
 
 /**
- * A means of repo-level synchronisation for select RepoIO operations.
+ * A means of repo-level synchronisation for select RepoIO operations. Only one instance of this class
+ * is available at any time availabel through {@code getRepoOpControl}. A new instance can be created with
+ * {@code createRepoOpControl} and will replace any existing instance
  */
-public class RepoOpControl {
+public final class RepoOpControl {
 
     private static final Logger logger = LogManager.getLogger(RepoOpControl.class.getName());
 
     private final RepoIO repoIO;
+    private final MultiModel models;
 
     private final ExecutorService pool = Executors.newCachedThreadPool();
     private final Map<String, BlockingQueue<RepoOp>> queues = new HashMap<>();
 
-    public RepoOpControl(RepoIO repoIO) {
+    private static Optional<RepoOpControl> repoOpControlOptional = Optional.empty();
+
+    private RepoOpControl(RepoIO repoIO, MultiModel models) {
         this.repoIO = repoIO;
+        this.models = models;
+    }
+
+    /**
+     * Gets an RepoOpControl instance that has been created previously by {@code createRepoOpControl}
+     * @return
+     */
+    public static RepoOpControl getRepoOpControl() {
```

Comment:
The simplest way (barring making everything singletons) is probably like this:
```java
A a = new A();
B b = new B();
a.setB(b);
b.setA(a);
```
I think the 'right' way is to not have RepoIO depend on RepoOpControl though.